src/hooks/useCheckURL.ts

```diff
@@ -41,7 +44,88 @@ function useCheckURL(urlToCheck: string): {
 			throw new Error("User handle could not be extracted from keystore");
 		}
 		const u = new URL(urlToCheck);
-		if (u.protocol === 'openid-credential-offer' || u.searchParams.get('credential_offer') || u.searchParams.get('credential_offer_uri') ) {
+		if (u.searchParams.get('credential_offer_uri') ) {
+			const credentailOfferResponse = await fetch(u.searchParams.get('credential_offer_uri'));
+			const credentialOffer = await credentailOfferResponse.json();
+
+			if (!credentialOffer.grants['urn:ietf:params:oauth:grant-type:pre-authorized_code'] || !credentialOffer.credential_issuer) {
+				return;
+			}
+
+			const body = {
+				grant_type: 'urn:ietf:params:oauth:grant-type:pre-authorized_code',
+				'pre-authorized_code': credentialOffer.grants['urn:ietf:params:oauth:grant-type:pre-authorized_code']['pre-authorized_code'],
+			};
+
+			const tokenResponse = await fetch(`${credentialOffer.credential_issuer}/token`, {
+				method: 'POST',
+				body: JSON.stringify(body),
+				headers: {
+					'Content-Type': 'application/json',
+				}
+			});
+			const { access_token: accesToken, c_nonce: cNonce } = await tokenResponse.json();
+
+			const openIdCredentialIssuerResponse = await fetch(`${credentialOffer.credential_issuer}/.well-known/openid-credential-issuer`, {
+				headers: {
+					'Cache-Control': 'no-cache',
+				},
+			});
+			const openIdCredentialIssuer = await openIdCredentialIssuerResponse.json();
+			// const metadata = OpenidCredentialIssuerMetadataSchema.parse(openIdCredentialIssuer);
+			const metadata = openIdCredentialIssuer; // @todo: investigate schema errors
+
+			const generateNonceProof = async (cNonce: string, audience: string, clientId: string): Promise<{ jws: string }> => {
+				const [{ proof_jwt }] = await keystore.generateOpenid4vciProof(cNonce, audience, clientId);
+				return { jws: proof_jwt };
+			};
+
+			const storeCredential = async (c: StorableCredential) => {
+				await api.post('/storage/vc', {
+					...c
+				});
+			};
+
+			const credentialEndpoint = metadata.credential_endpoint;
+			const credentialRequestHeaders = {
+				'Authorization': `Bearer ${accesToken}`,
+				'Content-Type': 'application/json',
+			};
+
+			const { jws } = await generateNonceProof(cNonce, metadata.credential_issuer, 'wwwallet');
+
+			const credentialEndpointBody = {
+				'proof': {
+					'proof_type': 'jwt',
+					'jwt': jws,
+				},
+				format: VerifiableCredentialFormat.JWT_VC_JSON, // @todo: retrieve from credential_configurations_supported
```

Comment:
`credential_configurations_supported` is an array of one or more supported formats. A chain pattern may be appropriate for deciding which supported format to use, since this affects subsequent processing of the credential.